### PR TITLE
Add functionality from suggestion plugin

### PR DESCRIPTION
**Pull request by micgro42**
Adding functionality from the undocumented suggestion plugin.

The functionality to send an email both when a new draft is created and when one is approved is implemented.

The functionality to send an email when a draft is rejected is not implemented because of the following reasons:
  * There is no reject functionality in the current publish plugin
  * Unclear priorities when a draft is approved by 2 of 3 reviewers and then rejected by another one
  * Unclear what is supposed to happen if an old draft is rejected.
  * The preferred course of action with suboptimal drafts is to comment on how they can be improved and not their rejection

### CODE REVIEW

**Comment by micgro42** [comment]
Mh, setting the email-adresse from the person doing the approving as FROM triggers many phishing/spam filters. This should be improved

**Comment by splitbrain** [comment]
Could you keep the original From but add a Reply-To header?

File: action/approve.php [hunk 1]
```diff
@@ -74,6 +74,10 @@ function addApproval() {
             $data['id'] = $ID;
             $data['approver'] = $_SERVER['REMOTE_USER'];
             $data['approver_info'] = $USERINFO;
+            if ($this->getConf('send_mail_on_approve') && $this->helper->isRevisionApproved($approvalRevision)) {
+                $mail =  new action_plugin_publish_mail();
```
Comment (splitbrain) [inline action/approve.php]: the recommended way to load plugins (and components) is to use plugin_load().

File: action/mail.php [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+        $controller->register_hook('IO_WIKIPAGE_WRITE', 'AFTER', $this, 'send_change_mail', array());
+    }
+
+    /**
+     * send an email to inform about a changed page
+     *
+     * @param $event
+     * @param $param
+     * @return bool|mixed
```
Comment (splitbrain) [inline action/mail.php]: some short description about the parameters would be helpful. especially for the return value here

File: action/mail.php [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            return true;
+        }
+
+        //are we supposed to send change-mails at all?
+        if (!$this->getConf('send_mail_on_change')) {
+            return true;
+        }
+
+        // get mail receiver
+        $receiver = $this->getConf('apr_mail_receiver');
```
Comment (splitbrain) [inline action/mail.php]: Could send_mail_on_change be true but apr_mail_receiver be not set? Would it make sense to just have one setting (apr_mail_receiver which would be empty to disable the feature).

File: action/mail.php [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+            dbglog('[publish plugin]: Mail not send. Sender is admin.');
+            return true;
+        }
+
+        // get mail subject
+        $timestamp = $data['lastmod'];
+        $datum = dformat("d.m.Y",$timestamp);
+        $uhrzeit = dformat("H:i",$timestamp);
+        $subject = $this->getLang('apr_mail_subject') . ': ' . $ID . ' - ' . $datum . ' ' . $uhrzeit;
+        dbglog($subject);
```
Comment (splitbrain) [inline action/mail.php]: If you want to keep all the debug statements in, at least prefix them all with the same prefix ([publish plugin]).

File: action/mail.php [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+        $body = $this->create_mail_body('change');
+
+        dbglog('mail_send?');
+
+        $replyto_header = mail_encode_address($sender, 'Reply-To');
+        $sender = 'DokuWiki <www-data@127.0.0.1>';
+
+        $returnStatus = mail_send($receiver, $subject, $body, $sender,"","",$replyto_header);
+        dbglog($returnStatus);
```
Comment (splitbrain) [inline action/mail.php]: see my other comments on 3c21cb4927322d5890419a4dfcebc22ebbdb0f42

File: action/mail.php [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+        $body = $this->create_mail_body('approve');
+
+        $replyto_header = mail_encode_address($sender, 'Reply-To');
+        $sender = 'DokuWiki <www-data@127.0.0.1>';
+
+        return mail_send($receiver, $subject, $body, $sender,"","",$replyto_header);
+    }
+
+    /**
+     * erzeugt den Link auf die edit-Seite
```
Comment (splitbrain) [inline action/mail.php]: wrong language

File: conf/default.php [hunk 7]
```diff
@@ -5,7 +5,12 @@
...
 $conf['hide drafts'] = 0;
-$conf['hide approved banner'] = 0;
+$conf['hide_approved_banner'] = 0;
 $conf['author groups'] = '';
 $conf['internal note'] = '';
 $conf['delete attic on first approve'] = 0;
+
+$conf['send_mail_on_change'] = 0;
+$conf['send_mail_on_approve'] = 0;
+$conf['apr_mail_receiver'] = 'jirasmtp@localhost';
```
Comment (splitbrain) [inline conf/default.php]: this should not have a default

File: lang/de/lang.php [hunk 8]
```diff
@@ -1,24 +1,56 @@
...
+$lang['mail_greeting'] = 'Hallo @FULLNAME@!';
+
+$lang['mail_new_suggestiopns'] = 'Ein neuer Vorschlag für @TITLE@ auf @DOKUWIKIURL@ wurde erstellt.
+
+Ansehen and Bestätigen: @URL@';
+
+$lang['mail_changes_to_approved_rev'] = 'Änderungen zur letzten bestätigten Version: @CHANGES@';
+$lang['mail_changes_to_previous_rev'] = 'Änderungen zur vorherigen Version: @CHANGES@';
+
+$lang['mail_approved'] = 'Ihr Vorschlag für @TITLE@ auf @DOKUWIKIURL@ wurde bestätigt.
```
Comment (splitbrain) [inline lang/de/lang.php]: these mail bodies should be in separate text files in the lang directory. They can be accessed via $this->localFN() then.

File: lang/de/lang.php [hunk 9]
```diff
@@ -1,24 +1,56 @@
...
+
+URL: @URL@';
+
+
+
+$lang['mail_dw_signature'] =
```
Comment (splitbrain) [inline lang/de/lang.php]: the signature should be part of the mail body until splitbrain/dokuwiki/pull/1101 is merged

**Pull request merged**